src/libraries/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.UnitTests/System/Runtime/InteropServices/Marshal/ThrowExceptionForHRTests.cs

```diff
@@ -94,12 +95,58 @@ public void ThrowExceptionForHR_InvalidHR_Nop(int errorCode)
             Marshal.ThrowExceptionForHR(errorCode, IntPtr.Zero);
         }
 
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsBuiltInComEnabled))]
+        public void ThrowExceptionForHR_BasedOnISupportErrorInfo()
+        {
+            var comWrappers = new StrategyBasedComWrappers();
+            Guid iid = new Guid("999b8152-1e6f-4166-8b35-ac89475e96fa");
+            var obj = new ConditionallySupportErrorInfo(iid);
+            IntPtr pUnk = comWrappers.GetOrCreateComInterfaceForObject(obj, CreateComInterfaceFlags.None);
+            try
+            {
+                var exception = new InvalidOperationException();
+                ClearCurrentIErrorInfo();
+
+                // Set the error info for the current thread to the exception.
+                _ = Marshal.GetHRForException(exception);
```

Comment:
This needs to be done again. The first call to `Marshal.GetExceptionForHR()` will clear the `IErrorInfo` from the thread.